plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -221,12 +221,19 @@ class GoogleBatchTaskHandler extends TaskHandler {
         final now = System.currentTimeMillis()
         final delta =  now - timestamp;
         if( !jobState || delta >= 1_000) {
-            def newState = client.getJobState(jobId)
+            final newStatus = client.getJobStatus(jobId)
+            final newState = client.getJobState(newStatus)
```

Comment:
This calls twice `describeJob` API, which would be better to avoid. It should be enough to use `getJobStatus` and then take the state from the result as done here 

https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-google/src/main/nextflow/cloud/google/batch/client/BatchClient.groovy#L106-L107

Reply:
The API is only called once, the `getJobState()` is just a helper function to get the state from the status object.

Reply:
I see it now, but in this case, does not make much sense to make it a client class method. It would be better to method in the GoogleBatchTaskHandler class

Reply:
Well this method is also used by BatchClient which is why I left it there

Reply:
how about this, I will move `getJobState(String jobId)` to the task handler as it is the convenience method